api/src/main/java/io/serverlessworkflow/api/deserializers/DataInputSchemaDeserializer.java

```diff
@@ -46,17 +53,44 @@ public DataInputSchema deserialize(JsonParser jp, DeserializationContext ctxt)
     JsonNode node = jp.getCodec().readTree(jp);
 
     DataInputSchema dataInputSchema = new DataInputSchema();
+    JsonNode schemaDefinition = null;
 
-    if (!node.isObject()) {
-      dataInputSchema.setSchema(node.asText());
-      dataInputSchema.setFailOnValidationErrors(true); // default
-
-      return dataInputSchema;
-    } else {
-      dataInputSchema.setSchema(node.get("schema").asText());
+    if (node.isObject() && node.get("schema").isObject() && !node.get("schema").isEmpty()) {
+      schemaDefinition = node.get("schema");
       dataInputSchema.setFailOnValidationErrors(node.get("failOnValidationErrors").asBoolean());
+    } else {
```

Comment:
hmmm, 
why are we not supporting this?
```
"dataInputSchema": {
      "title": "MyJSONSchema",
      "properties":{
        "firstName":{
          "type": "string"
        },
        "lastName":{
          "type": "string"
        }
      }
    }
```
And requiring the inlined schema to be inside an "schema" field.

Reply:
This should be changed in the spec: https://github.com/serverlessworkflow/specification/blob/main/schema/workflow.json#L48-L74

The spec requires the `schema` attribute. But yeah, I think it would be an excellent enhancement to the spec. Can you open an issue there?

Reply:
@tsurdilo 
After this PR we will be supporting this
```
"dataInputSchema": {
      "schema": {"title": "MyJSONSchema",
      "properties":{
        "firstName":{
          "type": "string"
        },
        "lastName":{
          "type": "string"
        }
      }
}
    }
```
and 
```
"dataInputSchema": {
  "filecontainingschema.json"
}
```
and
```
"dataInputSchema": {
  "schema": "filecontainingschema.json"
}
```

I think spec should be changed to also support this

```
"dataInputSchema": {
 {"title": "MyJSONSchema",
      "properties":{
        "firstName":{
          "type": "string"
        },
        "lastName":{
          "type": "string"
        }
      }
    }
```
There is an asymetry there., that should be corrected. A minor thing that is slightly related with this PR and should be done with a different one if we agree. Lets merge this one as it is